_universum/modules/vcs/perforce_vcs.py

```diff
@@ -107,7 +107,7 @@ def disconnect(self):
             self.p4.disconnect()
             if not w:
                 return
-            if "Not connected" in w[0].message.message:
+            if "Not connected" in str(w[0].message):
```

Comment:
Please check: this is a Perforce-specific warning, I'm not sure str() result is the same as .message

Reply:
*w[0].message* is exception object. This object have no *message* property anymore